### PR TITLE
More fixes for the host report

### PR DESCRIPTION
**Pull request by znarf**

### CODE REVIEW

File: reports/host-report.js [hunk 1]
```diff
@@ -231,13 +230,6 @@ async function HostReport(year, month, hostId) {
       let transactions = await getTransactions(Object.keys(collectivesById), startDate, endDate, {
         where: {
           HostCollectiveId: host.id,
-          kind: Object.values(
-            omit(TransactionKind, [
-              'HOST_FEE', // Host fee is loaded separately and added as a column
-              'HOST_FEE_SHARE', // Not surfaced yet, to keep the report as close to the previous version as possible
-              'HOST_FEE_SHARE_DEBT', // Not surfaced yet, to keep the report as close to the previous version as possible
-            ]),
-          ),
```
Comment (znarf) [inline reports/host-report.js:240]: We need this data so we should not ignore them at this point.

File: reports/host-report.js [hunk 2]
```diff
@@ -306,11 +291,12 @@ async function HostReport(year, month, hostId) {
           return 'donations';
         } else if (t.ExpenseId && t.type === 'DEBIT') {
           return 'expenses';
-          // TODO REPLACE WITH OTHER INCOMES AND OTHER EXPENSES
-        } else if (t.type === 'DEBIT') {
+        } else if (t.type === 'DEBIT' && t.kind !== null) {
           return 'otherDebits';
-        } else if (t.type === 'CREDIT') {
+        } else if (t.type === 'CREDIT' && t.kind !== null) {
```
Comment (znarf) [inline reports/host-report.js:296]: Ignoring the few transactions with kind=null

File: reports/host-report.js [hunk 3]
```diff
@@ -263,17 +261,11 @@ async function HostReport(year, month, hostId) {
       });
 
       if (!transactions || transactions.length === 0) {
-        throw new Error('No transaction found');
+        console.log(`No transaction found for ${host.slug}, skipping`);
```
Comment (Betree) [inline reports/host-report.js:264]: This is to clear the logs, the previous version was polluted with unnecessary stacktraces

image

**Comment by Betree** [comment]
Added the ability to add erratum message and number via env variables:


With the number (only affects title):


I'm also proposing the following message as Erratum for the latest host report issue:

> We're currently changing how we record transactions to make everything easier to understand, more standard, and scalable. As part of these changes, we missed an update in the previous "2021/6 Monthly Host Report" email that translated into showing wrong numbers in the breakdown below for Host Fees and Host Fee Shares.
> 
> This email includes the fixed numbers. Feel free to let us know if you see anything wrong or if you have any questions!

**Pull request merged**